npm/cmd/main.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2018 Microsoft. All rights reserved.
+// MIT License
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	"github.com/Azure/azure-container-networking/log"
+	npmconfig "github.com/Azure/azure-container-networking/npm/config"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"k8s.io/klog"
+)
+
+// Version is populated by make during build.
+var version string
+var cfgFile string
+
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:   "azure-npm",
+	Short: "Collection of functions related to Azure NPM's debugging tools",
+
+	RunE: func(cmd *cobra.Command, args []string) error {
+		config := &npmconfig.Config{}
+		err := viper.Unmarshal(config)
+		if err != nil {
+			fmt.Printf("unable to decode into config struct, %v", err)
+		}
+
+		Start(*config)
```

Comment:
similarly, this func should block or return an error, and we should `return Start(...)`